lib/route/cls/cgroup.c

```diff
@@ -32,15 +32,10 @@ static int cgroup_clone(void *_dst, void *_src)
 {
 	struct rtnl_cgroup *dst = NULL, *src = _src;
 
-	dst = calloc(1, sizeof(*dst));
-	if (!dst)
-		return -NLE_NOMEM;
-
-	dst->cg_mask = src->cg_mask;
-	dst->cg_ematch = rtnl_ematch_tree_clone(src->cg_ematch);
-	if (!dst) {
-		free(dst);
-		return -NLE_NOMEM;
+	if (src->cg_ematch) {
+		dst->cg_ematch = rtnl_ematch_tree_clone(src->cg_ematch);
+		if (!dst->cg_ematch)
+			return -NLE_NOMEM;
```

Comment:
this patch is somehow not right. It doesn't use the `_dst` parameter, but also does not initialized `dst` variable.

Reply:
Fixed.